tests/candidates/test_matchers.py

```diff
@@ -173,3 +196,231 @@ def test_inverse(doc_setup):
     # Check if Inverse raises an error when two child matchers are provided.
     with pytest.raises(ValueError):
         Inverse(matcher0, matcher0)
+
+
+def test_cancat(doc_setup):
+    """Test Concat matcher."""
+    doc = doc_setup
+    space = MentionNgrams(n_min=1, n_max=2)
+
+    # Match any span that contains "this"
+    matcher0 = RegexMatchSpan(
+        rgx=r"this", search=False, full_match=False, longest_match_only=False
+    )
+    # Match any span that contains "is"
+    matcher1 = RegexMatchSpan(
+        rgx=r"is", search=False, full_match=False, longest_match_only=False
+    )
+    matcher = Concat(matcher0, matcher1)
+    assert set(tc.get_span() for tc in matcher.apply(space.apply(doc))) == {"This is"}
+
+    # Test if matcher raises an error when _f is given non-TemporarySpanMention
+    with pytest.raises(ValueError):
+        list(matcher.apply(doc.sentences[0].words))
+
+    # Test if an error is raised when the number of child matchers is not 2.
+    matcher = Concat(matcher0)
+    with pytest.raises(ValueError):
+        list(matcher.apply(space.apply(doc)))
+
+    # Test with left_required=False
+    matcher = Concat(matcher0, matcher1, left_required=False)
+    assert set(tc.get_span() for tc in matcher.apply(space.apply(doc))) == {
+        "This is",
+        "is apple",
+    }
+
+    # Test with right_required=False
+    matcher = Concat(matcher0, matcher1, right_required=False)
+    assert set(tc.get_span() for tc in matcher.apply(space.apply(doc))) == {"This is"}
+
+    # Test with permutations=False
+    matcher = Concat(matcher1, matcher0, permutations=False)
+    assert set(matcher.apply(space.apply(doc))) == set()
+    # Test with permutations=True
+    matcher = Concat(matcher1, matcher0, permutations=True)
+    assert set(tc.get_span() for tc in matcher.apply(space.apply(doc))) == {"This is"}
+
+    # TODO: Add a test for ignore_sep=False
```

Comment:
Are we going to have this test? Thanks!

Reply:
I was not going to as I don't know how to use `ignore_sep=False`.
But let me see the code and try to implement it!

Reply:
This option dates back to https://github.com/snorkel-team/snorkel/issues/262 and https://github.com/snorkel-team/snorkel-extraction/commit/4eb2907f10fb8652fa16296e625ad1bc47e31b9e.

Reply:
I think I found a bug around options `sep` and `ignore_sep`, which is not easy to fix.
Let's keep it TODO here instead of writing tests that just pass and fool us.